### PR TITLE
upgrade squid bridge

### PR DESCRIPTION
**Pull request by hypery2k**
Issue #157 upgrade squid bridge

### CODE REVIEW

File: checkstyle-sonar-plugin/pom.xml [hunk 1]
```diff
@@ -141,6 +141,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+ 			<groupId>com.google.guava</groupId>
+ 			<artifactId>guava</artifactId>
+ 			<version>15.0</version>
```
Comment (rnveach) [inline checkstyle-sonar-plugin/pom.xml]: Will this cause issues with checkstyle as it uses guava 26?

**Comment by rnveach** [comment]
@hypery2k CI must be green.
> File 'pom.xml' uses tabs for indentation

@romani wercker error:
> [ERROR] SonarQube server [http://localhost:9000] can not be reached

**Comment by hypery2k** [comment]
did you have a formatter for IDEA?

**Comment by rnveach** [comment]
@hypery2k I'm not sure I understand your question. It isn't a formatter, it is just asking to use spaces instead of tabs for indentation. This should be a simple change using notepad or such. Just follow formatting of similar items around it.

**Comment by hypery2k** [comment]
I cannot get the CI build green, as either team city or checkstyle will complain about [this](https://teamcity.jetbrains.com/viewLog.html?buildId=1618925&tab=Inspection&buildTypeId=Checkstyle_SonarCheckstyleIdeaInspectionsPullRequest). And I didn't changed anything there...

**Comment by rnveach** [comment]
weird it was only complaining about the indentation before. It looks like it is related to the new TC version.

**Comment by aepfli** [comment]
is it possible to support somehow? i am not sure how to fix it, but at least i understand the problem in CheckstyleExecutor with "Call to 'Collection.toArray()' with zero-length array argument (1)" https://github.com/checkstyle/sonar-checkstyle/blob/010d4083c0e901148a6438a65a6a31f43fe95957/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleExecutor.java#L146

**Comment by rnveach** [comment]
>  would be to change the line simply to return new URLClassLoader(urls.toArray(new URL[0]), null);

Yes, this is the change that would have to be made.

**Comment by hypery2k** [comment]
with the proposed changes I then get maven pmd errors:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.7:check (default) on project checkstyle-sonar-plugin: You have 1 PMD violation. For more details see: /Users/mreinhardt/Documents/Programmierung/repositories/git/job/sonar-checkstyle/checkstyle-sonar-plugin/target/pmd.xml -> [Help 1]
```

**Comment by rnveach** [comment]
@hypery2k You would have to show us the PMD error as that is all the error shows. Most likely it isn't related to your PR but TC changed between the last master and now.

**Comment by hypery2k** [comment]
see here:
```
<?xml version="1.0" encoding="UTF-8"?>
<pmd version="5.5.1" timestamp="2018-09-24T16:27:44.057">
<file name="/Users/mreinhardt/Documents/Programmierung/repositories/git/job/sonar-checkstyle/checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleExecutor.java">
<violation beginline="146" endline="146" begincolumn="35" endcolumn="58" rule="OptimizableToArrayCall" ruleset="Design" package="org.sonar.plugins.checkstyle" class="CheckstyleExecutor" method="createClassloader" externalInfoUrl="https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/design.html#OptimizableToArrayCall" priority="3">
This call to Collection.toArray() may be optimizable
</violation>
</file>
</pmd>

```

**Comment by aepfli** [comment]
Well I do think that this pmd rule, based on the link I posted is not 100% valid ;) - I am not a contributor to sonar-checkstyle, but can we deactivate this rule?

**Comment by rnveach** [comment]
Main repo uses the same IDEA and PMD doesn't complain there, so I am trying to reconcile the 2. See https://github.com/checkstyle/sonar-checkstyle/pull/159 . Sorry for delay.

**Comment by romani** [comment]
@hypery2k , I apologize for delay in response.

Please squash all commits in one and rebase latest master (we recently merged some fixes for CI), and will continue investigate what else is required to make this PR be merged.

Please also make some comments in issue about what need to be changed in code to let us validate a fix. We need help from you.

**Comment by rnveach** [comment]
@hypery2k ping

**Comment by romani** [comment]
closed in favor of https://github.com/checkstyle/sonar-checkstyle/pull/165

**Pull request closed without merging**